src/Strategy/Ruby/Gemspec.hs

```diff
@@ -0,0 +1,108 @@
+module Strategy.Ruby.Gemspec (
+  rubyString,
+  parseRubyAssignment,
+  Assignment (..),
+  readAssignments,
+  parseRubyArray,
+  parseRubyWordsArray,
+  rubyLicenseValuesP,
+) where
+
+import Control.Applicative ((<|>))
+import Data.Char (isSpace)
+import Data.Functor (($>))
+import Data.List.Extra (singleton)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Void (Void)
+import Text.Megaparsec (MonadParsec (eof), Parsec, anySingle, anySingleBut, between, choice, lookAhead, many, optional, sepBy, skipManyTill, takeWhile1P, takeWhileP, try)
+import Text.Megaparsec.Char (char, space, space1, string)
+
+type Parser = Parsec Void Text
+
+-- |Given a single start delimiter, return start/end delimiters
+selectDelim :: Char -> (Char, Char)
+selectDelim = \case
+  '{' -> ('{', '}')
+  '<' -> ('<', '>')
+  '(' -> ('(', ')')
+  '[' -> ('[', ']')
+  c -> (c, c)
+
+rubyString :: Parser Text
+rubyString =
+  -- '.freeze' is a ruby idiom that turns a string into an immutable version of
+  -- itself. In this case it's safe to just ignore it and take the string it's
+  -- attached to.
+  toText <$> (stringText <* optional (string ".freeze"))
+  where
+    betweenDelim :: Char -> Parsec Void Text String
+    betweenDelim c =
```

Comment:
I'm not familiar with how this parsing library works, but does this deal with escaped delimiters within the string?

E.g. `"This is a \"string\" with quotes in it"` or `'we don\'t need to worry about escapes'`

There are similar worries for arbitrary delimiters: `%Q{this string \{keeps on going\} until here}`

This isn't too likely for the data-set we're looking at here (hopefully SPDX expressions), but if it's easy to deal with it'd be nice.

Reply:
Also, for strings delimited by `"` or strings with arbitrary delimiters that start with `%Q`, you can have string interpolation like this:

```
"hello, #{name}"
%Q{hello, #{name}}
```

Should we just fail if we see string interpolation?

Reply:
I've added support for escaping the closing delimiter in strings. In all my tests I didn't see this situation but it's good to have.